### DISCUSSION TITLE
Understanding multi-platform support

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#325] [Q&A] [answered]
I'm trying to understand how different platforms are targeted by SK. I've been using it for HL2 (and thus UWP) applications and it was a very smooth development experience in C# so far. I know there's support for other devices and platforms, like the Quest, but I'm unclear if those can be targeted from the same C# code, or if that requires dropping down to the C++ level libraries in SK? Is there a development workflow tutorial somewhere for targeting the Quest, as an example (as I'd really like to try out the pass-through support)?

Also, the "Linux support" mentioned in the README appears to be referencing devices _running_ Linux, right? And not _developing_ SK applications on Linux, which then get deployed to whatever device. Just trying to make sure I understand the basics of how the C++ and C# code fits together with the different target platforms and related development workflow.

**Comment by maluoi** [comment, marked as answer]
Here's the current platform support matrix! (as of v0.3.5)
| Platform | Project Type | Simulator | Notes |
|---|---|---|---|
| Windows Holographic | .Net UWP / C | ✅ | |
| Windows Desktop | .Net UWP / .Net Core / C | ✅ | |
| Linux | .Net Core / C| ✅ | |
| Android | Xamarin / C | | OpenXR loader is currently Quest specific |
| Mac | | | Can only deploy to Android via Xamarin |
| Web | C | ✅ | Simulator _only_ right now |

## Cross Platform Template
The big thing you're probably interested in here is the [cross platform template](https://github.com/maluoi/StereoKit/tree/master/StereoKitTemplates/StereoKitXPlat), which supports .Net Core, .Net UWP, and Xamarin all at once. There's currently not a standalone Xamarin template, since Xamarin only deploys to Android, and that would not allow for simulator development.

The cross platform template is unfortunately not documented super well because it's not exactly what I want to ship! It's a little complex, and is composed of at least 4 different Visual Studio Projects in a single Solution. The multi-project situation means it's either difficult, or impossible to ship as a proper template (I'm not certain it's impossible, but I haven't found a way to do it yet). **However**, there's still a number of people using it with plenty of success, so it'll totally work.

## .Net MAUI
What I _really_ want to ship is a template based on .Net Maui. Maui seems like it's intended to be the nice new successor to Xamarin, and it's capable of deploying to Desktop / Android / Mac / Web all from a _single_ project file. That sounds like a pretty good experience to me, and means I can make a proper template out of it!

Unfortunately, it's not done yet. I've looked at it, and I like it, and I'm really hoping it doesn't take too long.

A notable omission from Maui is lack of UWP support :( That won't be particularly fun for HoloLens, basically UWP is deprecated at this point, and is no longer receiving any new features or inclusion in this sort of thing.

**Reply by paulmelis** [reply]
Thanks a lot for the detailed answer!
